src/common/downloadFileDialog.component.tsx

```diff
@@ -0,0 +1,110 @@
+import WarningIcon from '@mui/icons-material/Warning';
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  FormHelperText,
+} from '@mui/material';
+import { useQueryClient } from '@tanstack/react-query';
+import { AxiosError } from 'axios';
+import React from 'react';
+import { APIImage, APIImageWithURL } from '../api/api.types';
+import { getImageQuery } from '../api/images';
+import handleIMS_APIError from '../handleIMS_APIError';
+import { downloadFileByLink } from '../utils';
+
+export interface BaseDownloadFileProps {
+  open: boolean;
+  onClose: () => void;
+  fileType: 'Image' | 'Attachment';
+}
+
+export interface ImageDownloadDialogProps extends BaseDownloadFileProps {
+  fileType: 'Image';
+  file: APIImage;
+}
+
+export type DownloadFileProps = ImageDownloadDialogProps;
+
+const DownloadFileDialog = (props: DownloadFileProps) => {
+  const { open, onClose, fileType, file } = props;
+
+  const getFile = fileType === 'Image' ? getImageQuery : getImageQuery;
```

Comment:
```suggestion
  const getFile = fileType === 'Image' ? getImageQuery : getImageQuery;
```

`getFile` is not a descriptive enough name. Consider using a more specific name, such as `getDownloadFileQuery`. Additionally, this conditional statement is incorrect, you should set the other case explicitly or leave it as `undefined`. Alternatively, consider removing it entirely so it can be handled properly in the attachments PR.

`const getDownloadFileQuery =  getImageQuery`